src/infer_request.cc

```diff
@@ -1281,31 +1293,34 @@ InferenceRequest::ValidateBytesInputs(
   int64_t element_idx = 0;
   size_t remaining_element_size = 0;
 
-  size_t buffer_idx = 0;
-  const size_t buffer_count = input.Data()->BufferCount();
+  size_t buffer_next_idx = 0;
+  const size_t buffer_count = input.DataBufferCount();
 
   const char* buffer = nullptr;
   size_t remaining_buffer_size = 0;
   TRITONSERVER_MemoryType buffer_memory_type;
   int64_t buffer_memory_id;
 
   // Validate elements until all buffers have been fully processed.
-  while (remaining_buffer_size || buffer_idx < buffer_count) {
+  while (remaining_buffer_size || buffer_next_idx < buffer_count) {
     // Get the next buffer if not currently processing one.
     if (!remaining_buffer_size) {
       // Reset remaining buffer size and pointers for next buffer.
-      buffer = input.Data()->BufferAt(
-          buffer_idx++, &remaining_buffer_size, &buffer_memory_type,
-          &buffer_memory_id);
+      RETURN_IF_ERROR(input.DataBuffer(
+          buffer_next_idx++, (const void**)(&buffer), &remaining_buffer_size,
+          &buffer_memory_type, &buffer_memory_id));
       *expected_byte_size += remaining_buffer_size;
 
+      if (buffer_next_idx > buffer_count) {
```

Comment:
Isn't this already checked by the while loop conditional? The largest value of `buffer_next_idx` at the beginning of the loop will be `buffer_count-1`, after line 1310 will be `buffer_count` and this line will never be true.

Reply:
`buffer_next_idx > buffer_count` statement is true if `remaining_buffer_size>0` and `buffer_next_idx == buffer_count` at the top while loop condition check. This happens when the last string element content is in the last two buffers but the second buffer somehow does not exist.

Reply:
>This happens when the last string element content is in the last two buffers but the second buffer somehow does not exist.

Can you point me to the test case that illustrate this edge case? I can't see why the 2nd buffer will not exist while input data says it contains 2 buffer.

> if `remaining_buffer_size>0` and `buffer_next_idx == buffer_count`
if `remaining_buffer_size>0`, then line 1307 `if (!remaining_buffer_size)` is false and won't hit this line.

Reply:
You are right. This check is redundent.